### PR TITLE
Removed Navigator from codebase

### LINKED ISSUE #2895: [Code health] Remove Navigator
Continue #2889 and Remove Navigator completely

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2895 

@shobhitagarwal1612  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/MainActivity.kt [hunk 1]
```diff
@@ -100,16 +92,18 @@ class MainActivity : AbstractActivity() {
         showPermissionDeniedDialog(viewGroup)
       }
       MainUiState.OnUserSignedOut -> {
-        navigator.navigate(SignInFragmentDirections.showSignInScreen())
+        navHostFragment.navController.navigate(SignInFragmentDirections.showSignInScreen())
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/MainActivity.kt:95]: Let's create a helper private method `navigateTo` and use here and below

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt [hunk 2]
```diff
@@ -70,7 +71,7 @@ class OfflineAreasFragment : AbstractFragment() {
     super.onViewCreated(view, savedInstanceState)
     lifecycleScope.launch {
       viewModel.navigateToOfflineAreaSelector.collect {
-        findNavController().navigate(OfflineAreasFragmentDirections.showOfflineAreaSelector())
+        findNavController().navigate(R.id.offline_area_selector_fragment)
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt:74]: Why is this changed required?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt (hunk 2 above)
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt:74]: The earlier one is not working as expected. passing fragment id is more correct approach than passing action id.

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt:74]: Using nav actions over view IDs provides type safety, self-documentation of app flows through the nav graph, and make nav intents clearer and more maintainable. Why was `OfflineAreasFragmentDirections.showOfflineAreaSelector()` not working?

**Review by gino-m — APPROVED** [review]

**Review by shobhitagarwal1612 — APPROVED** [review]
